frontend/javascripts/oxalis/model/sagas/undo_saga.ts

```diff
@@ -386,6 +390,7 @@ export function* manageUndoStates(): Saga<never> {
         if (!action) {
           throw new Error("Unexpected action");
         }
+        shouldClearRedoState = true;
```

Comment:
Why does the redo state always need to be cleared, even for actions that are not added to the undo stack? :thinking: Should it have been moved to line 397 instead (so into the `if (addToUndo)` block)?

Reply:
Yes, valid question! I added a comment for that. Let me know if you agree with the reasoning. The comment:

```
// Even if addToUndo is false, we will clear the redo state. Otherwise,
// selecting a segment while being in an earlier state and then doing a redo operation
// would forget about the newly registered segment which is weird.
// Clearing the redo state after an explicit action (such as selecting a segment)
// seems more intuitive.
// Also note that a segment update as a consequence of an annotation operation
// is already accompanied by an empty redo stack (because the volume operation
// will have cleared that stack). So, in that case shouldClearRedoState is true,
// but essentially nothing will be cleared, anyway.
```

Reply:
Thank you for explaining. I agree with the reasoning, but am having some additional thoughts. I've actually wondered about this during the review, too, but shouldn't these other explicit segment update actions be undoable/redoable, then, too? It's a bit weird that something that clears the redo stack isn't undoable.
However, I also see the issue that the undo stack might get "crowded", but as long as only explicit user actions end up on the undo stack I would argue that's fine. Maybe in the future we can find a way to "highlight" where changes were undone or show a small description of what was just undone - which could help with that issue.

I don't think this needs to be fully discussed for this issue. The current state is fine to be merged.

Reply:
> I agree with the reasoning, but am having some additional thoughts. I've actually wondered about this during the review, too, but shouldn't these other explicit segment update actions be undoable/redoable, then, too? It's a bit weird that something that clears the redo stack isn't undoable.

Ok, convinced :) Especially, the fact that changing a segment color wasn't undoable disturbed me. I just pushed a commit which takes care of this. I think it makes the code more explicit / easier, too. Could you do test the feature again? I did so, too, obviously, but I don't know whether I'm not seeing the forest for the trees after all this :)